src/odemis/driver/technolution.py

```diff
@@ -0,0 +1,544 @@
+        if method == _METHOD_GET:
+            resp = requests.get(url, json=data, timeout=timeout)
+        elif method == _METHOD_POST:
+            resp = requests.post(url, json=data, timeout=timeout)
+
+        if resp.status_code != expected_status:
+            raise AsmApiException(url, resp, expected_status)
+
+        logging.info("Call to %s went fine, no problems occured\n" % url)
+
+        if raw_response:
+            return resp
+        elif method == _METHOD_POST:
+            return resp.status_code
+        elif method == _METHOD_GET:
+            return json.loads(resp.content)
+
+
+class EBeamScanner(model.Emitter):
+
+    def __init__(self, name, role, parent, **kwargs):
+        super(EBeamScanner, self).__init__(name, role, **kwargs)
+        self.parent = parent
+
+        # self._metadata[model.MD_PIXEL_SIZE] =
+
+        clockFrequency = self.parent.simpleASMCall(self.parent.server_URL + "/scan/clock_frequency", _METHOD_GET, 200)[
+            'frequency']
+        self.clockFrequency = model.IntVA(clockFrequency, unit='Hz')
+        self.clockPeriod = model.FloatVA(1 / self.clockFrequency.value, unit='s')
+        self._shape = model.TupleVA((6400, 6400), unit='px,px')
+        self.resolution = model.ResolutionVA((6400, 6400), ((10, 10), (7200, 7200)), setter=self._setResolution)
+        self.dwellTime = model.FloatContinuous(float(self.clockPeriod.value), (float(self.clockPeriod.value), 100000.0),
+                                               unit='s')
+        self.pixelSize = model.TupleContinuous((4, 4), range=((1, 1), (100000, 100000)), unit='px,px',
+                                               setter=self._setpixelSize)
+        self.rotation = model.FloatContinuous(0, range=(0, 2 * numpy.pi), unit='rad, cw=+')
+        self.scanFlyback = model.FloatVA(0, unit='s')
+        self.scanOffset = model.TupleContinuous((0.0, 0.0), range=((-10.0, -10.0), (10.0, 10.0)), unit='V,V')
+        self.scanGain = model.TupleContinuous((0.0, 0.0), range=((-10.0, -10.0), (10.0, 10.0)), unit='V,V')
+        self.scanDelay = model.TupleContinuous((0, 0), range=((1, 1), (100000, 100000)), unit='s,s',
+                                               setter=self._setScanDelay)
+
+        self._metadata[model.MD_PIXEL_SIZE] = self.pixelSize.value
+        self._metadata[model.MD_DWELL_TIME] = self.dwellTime.value
+
+    def _setResolution(self, input_resolution):
+        if input_resolution[0] == input_resolution[1]:
+            return input_resolution
+        else:
+            logging.warning("Non-square resolution entered, only square resolution sizes are supported")
+            return self.resolution.value
+
+    def _setpixelSize(self, input_resolution):
+        if input_resolution[0] == input_resolution[1]:
+            return input_resolution
+        else:
+            logging.warning("Non-square pixel size entered, only square pixel sizes are supported")
+            return self.pixelSize.value
```

Comment:
For VAs on the hardware component, typically we try to accomodate at best what the request was. So here I'd suggest to just pick either X or Y, and duplicate it (instead of using the previous value).